packages/extension-huggingface/README.md

```diff
@@ -94,3 +96,60 @@ SELECT {{ products.value() | huggingface_table_question_answering(query=question
   }
 ]
 ```
+
+### Text Generation
+
+The [Text Generation](https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task) is one of the Natural Language Processing tasks supported by Hugging Face.
+
+Using the `huggingface_text_generation` filter. The result will be a string from `huggingface_text_generation`.
+
+**Notice**: The **Text Generation** default model is **gpt2**, If you would like to use the [Meta LLama2](https://huggingface.co/meta-llama) models, you have two method to do:
+
+1. Subscribe to the [Pro Account](https://huggingface.co/pricing#pro). Set the Meta LLama2 model using the `model` keyword argument in `huggingface_text_generation`, e.g: `meta-llama/Llama-2-13b-chat-hf`.
+
+**SQL**:
+
+```sql
+{% set data = [
+  {
+    "rank": 1,
+    "institution": "Massachusetts Institute of Technology (MIT)",
+    "location code":"US",
+    "location":"United States"
+  },
+  {
+    "rank": 2,
+    "institution": "University of Cambridge",
+    "location code":"UK",
+    "location":"United Kingdom"
+  },
+  {
+    "rank": 3,
+    "institution": "Stanford University"
+    "location code":"US",
+    "location":"United States"
+  }
+  -- other universities.....
+] %}
+
+SELECT {{ data | huggingface_text_generation(query="Which university is the top-ranked university?", model="meta-llama/Llama-2-13b-chat-hf") }} as result
+```
+
+**Response**:
+
+```json
+[
+  {
+    "result": "Answer: Based on the provided list, the top-ranked university is Massachusetts Institute of Technology (MIT) with a rank of 1."
+  }
+]
+```
+
+2. Select one of the [Meta LLama2](https://huggingface.co/meta-llama) Models and deploy it to the [Inference Endpoint](https://huggingface.co/inference-endpoints). Set the endpoint URL using the `endpoint` keyword argument in `huggingface_text_generation`.
+
+```sql
+SELECT {{ data | huggingface_text_generation(query="Which university is the top-ranked university?", endpoint='xxx.yyy.zzz.huggingface.cloud') }} as result
```

Comment:
Maybe we can merge these to code snippet and use "comment" to describe the detail.
I think it will be more readable.

Reply:
According to the code snippet, the marked code is older code. After discussion with @onlyjackfrost and checking, no need to change.